proposals/Layer-1.md

```diff
@@ -0,0 +1,390 @@
+the stack is an instance of the corresponding tagged exception type (defined by
+`except_index`).
+
+If the query succeeds, the data values (associated with the type signature of
+the exception class) are extracted and pushed onto the stack, and control
+transfers to the instructions in the then block.
+
+If the query fails, it either enters the else block, or transfer control to the
+end of the if_except block if there is no else block.
+
+### Debugging
+
+Earlier discussion implied that when an exception is thrown, the runtime will
+pop the operand stack across function calls until a corresponding, enclosing try
+block is found. The implementation may actually not do this. Rather, it may
+first search up the call stack to see if there is an enclosing try. If none are
+found, it could terminate the thread at the point of the throw. This would
+allow better debugging capability, since the corresponding call stack is still
+there to query.
+
+## Changes to the text format.
+
+This section describes change in the
+[instruction syntax document](https://github.com/WebAssembly/spec/blob/master/document/core/instructions.rst).
+
+### New instructions
+
+The following rules are added to *instructions*:
+
+```
+  try resulttype instructions* catch instructions* end |
+  except except_index |
+  throw |
+  if_except resulttype except_index then Instructions* end |
+  if_except resulttype except_index then Instructions* else Instructions* end
+```
+
+Like the `block`, `loop`, and `if` instructions, the `try` and `if_except`
+instructions are *structured* control flow instructions, and can be
+labeled. This allows branch instructions to exit try and `if_except` blocks.
+
+The `except_index` of the `except` and `if_except` instructions defines the
+exception type to create/extract form. See [exception index
+space](#exception-index-space) for further clarification of exception tags.
+
+## Changes to Modules document.
+
+This section describes change in the
+[Modules document](https://github.com/WebAssembly/design/blob/master/Modules.md).
+
+### Exception index space
+
+The _exception index space_ indexes all imported and internally-defined
+exceptions, assigning monotonically-increasing indices based on the order
+defined in the import and exception sections. Thus, the index space starts at
+zero with imported exceptions followed by internally-defined exceptions in
+the [exception section](#exception-section).
+
+## Changes to the binary model
+
+This section describes changes in
+the
+[binary encoding design document](https://github.com/WebAssembly/design/blob/master/BinaryEncoding.md).
+
+
+### Data Types
+
+#### except_ref
+
+An exception reference pointing to an instance of an exception. The size
+is fixed, but unknown in WebAssembly (the host defines the size in bytes).
+
+### Language Types
+
+| Opcode | Type constructor |
+|--------|------------------|
+| -0x41  |  `except_ref`    |
```

Comment:
I'd make this `0x7B`, since `except_ref` feels more like a [value type](https://webassembly.github.io/spec/core/binary/types.html#value-types) to me.

Reply:
Well, anyfunc will become a value type as well eventually. I think it makes sense to put all the reference types close together.